pkg/broker/ingress/handler.go

```diff
@@ -127,15 +136,26 @@ func (h *Handler) ServeHTTP(response nethttp.ResponseWriter, request *nethttp.Re
 
 	event.SetExtension(EventArrivalTime, cev2.Timestamp{Time: time.Now()})
 
+	ctx, span := trace.StartSpan(ctx, kntracing.BrokerMessagingDestination(broker))
```

Comment:
Should this check for an existing distributed trace? For example, if something has been replied back into the Broker. Or if the source that sent this already had started a trace.

Reply:
The existing trace context should always be set in the HTTP tracecontext so there is no need to check for the extension.